routes/car.js

```diff
@@ -1,8 +1,81 @@
 const express = require('express');
 const router = express.Router();
+const Car = require('../models/car');
+const User = require('../models/user');
+const jwtDecode = require('jwt-decode');
 
 router.get('/', (req, res) => {
-  res.send('car Route is working')
+    res.send('car Route is working')
 });
 
+
+router.post('/add', async (req, res) => {
+    const payload = jwtDecode(req.header('authorization'));
+    const update = {$setOnInsert: Object.assign({createdBy: payload['user']['_id']}, req.body)}
+    await Car.findOneAndUpdate({createdBy: payload['user']['_id'], license: req.body.license}, update,
```

Comment:
It's not findOneAndUpdate, you should be create a new Car document. and add the ObjectId into user.cars


> const user = await User.findById(userId)
> const car = new Cars({...})
> user.cars.push(car.id)

Reply:
Oh I see, so just to clarify, the intention is to store the `car` document as a nested object in the `user` collection

Reply:
Yes, but in user document,   user.cars is in form of array of strings. Then you can use mongoose populate when you need the cars info when query user
https://mongoosejs.com/docs/populate.html